src/dataflow/src/render/reduce.rs

```diff
@@ -26,6 +27,13 @@ use repr::{Datum, Row, RowArena, RowPacker};
 use super::context::Context;
 use crate::render::context::Arrangement;
 
+#[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
```

Comment:
I think my biggest problem with this and with the pr overall is that it adds a lot of complexity in a unorganized way. not sure how to best rectify that